openbas-framework/src/main/java/io/openbas/utils/schema/SchemaUtils.java

```diff
@@ -213,4 +213,13 @@ public static List<PropertySchema> getFilterableProperties(List<PropertySchema>
   public static List<PropertySchema> getSortableProperties(List<PropertySchema> propertySchemas) {
     return propertySchemas.stream().filter(PropertySchema::isSortable).collect(Collectors.toList());
   }
+
+  public static boolean isClassInPackage(String basePackage, String className) {
+    try {
+      Class<?> clazz = Class.forName(className);
+      return clazz.getPackageName().equals(basePackage);
+    } catch (ClassNotFoundException e) {
+      return false;
+    }
+  }
```

Comment:
It's a bit wasteful whenever the ctor of the given class does any kind of heavy lifting. Why not make a gatekeeping instance provider that does this exact validation and returns any successfully instantiated object?

Reply:
I'm not sure I understand what you're suggesting.

Reply:
This instantiates the class but throws the instance away after it's been successful; calling code then re-instantiates the class a second time to use it.

My suggestion is to create a helper that either returns the instance or throws as shown here, instead of just returning a boolean. What do you think?

Reply:
Oh, I see !
Good point, I will do it.

Reply:
It's not super important, it's not a memory or compute risk.

Reply:
I made the changes.